repos/jsutils/src/array/omitElement.js

```diff
@@ -0,0 +1,24 @@
+/** @module array */
+
+import { identity } from '../method/identity'
+
+/**
+ * Returns a new array containing all the elements of `array`, except for `element`
+ * @function
+ * @param {Array<*>} array
+ * @param {*} element
+ * @param {Function?} selector - optional fn returning a property of an element to compare for omission
+ * @returns {Array} subset of array without element
+ */
+export const omitElement = (array, element, selector=identity) => {
```

Comment:
It's really just a small shortcut:
```js
const arr = [1,2,3,4,5]
const result = omitElement(arr, 4)
result === [ 1, 2, 3, 5 ]
```
vs.
```js
```

Reply:
I mainly did it because I started to get into a pattern of passing around a selector function in the code surrounding the set hooks (like `useSet`), and I thought:

```js
omitElement(arr, element, selector)
```

was simpler than:

```js
arr.filter(x => selector(x) !== selector(element))
```

But now that you point this out, I see that it's really not much of a shortcut. I can remove it if that's preferred.

Reply:
gotcha. Personally, I would just use the filter prototype even if this were available for me to use. Since the shortcut doesnt really seem to give any other benefit besides slightly less letters to type.  with the addition of it being an extra dependency + us having to maintain it

Reply:
yeah agreed. I'll remove it 👍

Reply:
updated